File: server/shoppe/jv-service/src/main/java/com/lcaohoanq/jvservice/component/LocalizationUtils.java
package com.lcaohoanq.jvservice.component;

import com.lcaohoanq.jvservice.util.WebUtil;
import jakarta.servlet.http.HttpServletRequest;
import java.util.Locale;
import lombok.RequiredArgsConstructor;
import org.springframework.context.MessageSource;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.LocaleResolver;

@Component
@RequiredArgsConstructor
public class LocalizationUtils {

    private final MessageSource messageSource;
    private final LocaleResolver customLocaleResolver;

    public String getLocalizedMessage(String messageKey, Object ...params){
        HttpServletRequest request = WebUtil.getCurrentRequest();
        Locale locale = customLocaleResolver.resolveLocale(request);
        return messageSource.getMessage(messageKey, params, locale);
    }

}

File: spcserver/kotlin/src/main/kotlin/com/lcaohoanq/kotlinbasics/component/LocalizationUtils.kt
package com.lcaohoanq.kotlinbasics.component

import com.lcaohoanq.kotlinbasics.WebUtil
import jakarta.servlet.http.HttpServletRequest
import org.springframework.context.MessageSource
import org.springframework.stereotype.Component
import java.util.Locale


@Component
class LocalizationUtils(
    private val messageSource: MessageSource,
    private val customLocaleResolver: org.springframework.web.servlet.LocaleResolver
) {

    fun getLocalizedMessage(messageKey: String, vararg params: Any?): String {
        val request: HttpServletRequest = WebUtil.currentRequest
        val locale: Locale = customLocaleResolver.resolveLocale(request)
        return messageSource.getMessage(messageKey, params, locale)
    }
}
